### ISSUE TITLE
Question regarding unknown attributes mapping in Biolord reproduction

### ISSUE THREAD

**Issue by Peg-Wu** [nitzanlab/biolord_reproducibility#3]
Thank you for your excellent work.

I recently successfully reproduced Biolord on the drug perturbation prediction task (https://github.com/Peg-Wu/PertHub). However, I am a bit confused about the mapping of unknown attributes. 

During training, the latent representation of the unknown attributes for each cell is obtained via nn.Embedding using the cell's index in the entire h5ad file. The same applies during inference. For testing, the indices of control cells from the validation set are used. In theory, during the training process, only the embedding vectors corresponding to the indices of the training set cells are updated, while those corresponding to the validation set cells remain unupdated.

I don't quite understand why random indices from the training set aren't used directly during validation and testing instead. Is there any specific consideration behind this design?

I would appreciate your insights on this. Thank you!

Best regards,
Peng
